### PR TITLE
Add end to end tests to the 'deleteTeamsCmdF' mmctl command

### PR DESCRIPTION
**Pull request by gupsho**
#### Summary
Add end to end tests to the 'deleteTeamsCmdF' mmctl command

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15675

### CODE REVIEW

File: commands/team_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestDeleteTeamsCmdF() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForAllClients("Error deleting team which does not exist", func(c client.Client) {
+		printer.Clean()
+		teamName := "existingName"
```
Comment (streamer45) [inline commands/team_e2e_test.go:18]: As suggested in a previous PR, I'd set `teamName` to `nonExistentName` to make it more explicit we are looking for something that doesn't exist.

File: commands/team_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		cmd := &cobra.Command{}
+		args := []string{""}
+		args[0] = teamName
```
Comment (streamer45) [inline commands/team_e2e_test.go:21]: We can initialize and assign in the same line here (and in the following tests too):

```
args := []string{teamName}
```

File: commands/team_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		cmd.Flags().String("display_name", "newDisplayName", "Team Display Name")
+		cmd.Flags().Bool("confirm", true, "")
+
+		_ = deleteTeamsCmdF(s.th.Client, cmd, args)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Require().Equal("Unable to delete team '"+s.th.BasicTeam.Name+"' error: : You do not have the appropriate permissions., ", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Delete a valid team", func() {
```
Comment (streamer45) [inline commands/team_e2e_test.go:45]: I think we can run this using `RunForSystemAdminAndLocal`.

**Review by streamer45 — COMMENTED** [review]
Thank you @gupsho :+1: 
I've left a couple of minor suggestions but otherwise looks good!

File: commands/team_e2e_test.go (hunk 3 above)
Comment (gupsho) [reply commands/team_e2e_test.go:45]: I wasnt able to get it working for SystemAdminClient. This is error I get when I run with SystemAdminClient.
Unable to delete team 'faketeamaxkfw6' error: : Permanent team deletion feature is not enabled. Please contact your System Administrator.

File: commands/team_e2e_test.go (hunk 1 above)
Comment (gupsho) [reply commands/team_e2e_test.go]: Done.

File: commands/team_e2e_test.go (hunk 2 above)
Comment (gupsho) [reply commands/team_e2e_test.go]: Done.

File: commands/team_e2e_test.go (hunk 3 above)
Comment (streamer45) [reply commands/team_e2e_test.go:43]: Interesting, this is probably cause of [EnableAPITeamDeletion](https://docs.mattermost.com/administration/config-settings.html#enable-api-team-deletion) not being enabled on the server, which is the default.
Maybe we could add a test to check for this specific error when running as a System Admin.

Comment (gupsho) [reply commands/team_e2e_test.go:85]: Added.

**Review by streamer45 — APPROVED** [review]
Nicely done @gupsho :+1: Looks great!

File: commands/team_e2e_test.go (hunk 3 above)
Comment (mgdelacroix) [reply commands/team_e2e_test.go:85]: I think we could use `RunForSystemAdminAndLocal` in this test previously enabling API team deletion, and then explicitly disable it on the next test case and test how `s.th.SystemAdminClient` can't remove the team but `s.th.LocalClient` can.

I'm thinking on something like:

```go
s.RunForSystemAdminAndLocal("With API team deletion enabled, both sysadmin and local should be able to delete a team", func(c client.Client) {
    // create team
    // update config to enable API deletion
    // delete team and ensure it's gone
})

s.Run("With API team deletion disabled, only local client is able to delete a team", func() {
    // create team
    // update config to disable API deletion
    // run delete command with sysadmin and check that team is still there
    // run same command with local and check that the team is gone
})
```

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Looking really good @gupsho!! Thanks for the contribution!!

Added a couple requests to complete the test cases

File: commands/team_e2e_test.go (hunk 3 above)
Comment (gupsho) [reply commands/team_e2e_test.go]: Added.

File: commands/team_e2e_test.go [hunk 4]
```diff
@@ -51,3 +51,72 @@ func (s *MmctlE2ETestSuite) TestRenameTeamCmdF() {
...
+
+		cmd := &cobra.Command{}
+		args := []string{s.th.BasicTeam.Name}
+		cmd.Flags().String("display_name", "newDisplayName", "Team Display Name")
+		cmd.Flags().Bool("confirm", true, "")
+
+		_ = deleteTeamsCmdF(s.th.Client, cmd, args)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Require().Equal("Unable to delete team '"+s.th.BasicTeam.Name+"' error: : You do not have the appropriate permissions., ", printer.GetErrorLines()[0])
```
[the remark this answers is not in the document]
Comment (gupsho) [reply commands/team_e2e_test.go:84]: Done.

**Comment by gupsho** [comment]
> Looking really good @gupsho!! Thanks for the contribution!!
> 
> Added a couple requests to complete the test cases

Done.

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks a lot @gupsho for applying the changes!

**Review by lindy65 — APPROVED** [review]
Thanks @gupsho @mgdelacroix - looks good to me 👍

**Comment by jfrerich** [comment]
Thanks for the PR, @gupsho!  If you could resolve the merge conflict we will get this merged!

**Comment by gupsho** [comment]
> Thanks for the PR, @gupsho! If you could resolve the merge conflict we will get this merged!

Done.

**Comment by hanzei** [comment]
Thank you very much for the contribution @gupsho :+1:

**Pull request merged**